crates/cheatnet/src/state.rs

```diff
@@ -222,3 +216,13 @@ impl CheatnetState {
             || self.address_is_spoofed(contract_address)
     }
 }
+
+fn match_node_response<T: Default>(result: StateResult<T>) -> StateResult<T> {
+    match result {
+        Ok(class_hash) => Ok(class_hash),
+        Err(StateError::StateReadError(msg)) if msg.contains("node") => {
+            Err(StateError::StateReadError(msg))
+        }
+        _ => Ok(Default::default()),
```

Comment:
Why do we return default here?

Reply:
Because we have to

Reply:
Imagine a situation when someone wants to deploy a new contract when forking.
The state is asked this way:
- dict_state_reader (which does not have a value)
- fork_state_reader:
  - node is fetched and if there is no value, an error is returned
  - we have to return default because blockifier handles the state this way